configure.ac

```diff
@@ -125,6 +127,14 @@ AS_IF([test -n "$missing_leveldb"],
       [AC_MSG_ERROR([could not find the leveldb/snappy libraries])])
 LIBS="$save_LIBS"
 
+save_LIBS="$LIBS"
+AS_UNSET([LIBS])
+AC_SEARCH_LIBS([sqlite3_open], [sqlite3],, [missing_sqlite3=1], [$save_LIBS])
+AC_SUBST([sqlite3_LIBS], [$LIBS])
+AS_IF([test -n "$missing_sqlite3"],
+      [AC_MSG_ERROR([could not find the sqlite3 library])])
```

Comment:
If I'm not mistaken, that just causes a message to be emitted, but doesn't stop it, right? In other words, SQLite is optional? Is that what we want? I *think* so, but just checking... :smiley:

Reply:
This will stop `./configure`, so implies SQLite is required.  Builds using gclient will always install sqlite3, so I'm hoping that's OK.

Reply:
OK, that's not what I thought, and I thought we could build without SQLite? I think I'm wrong, though, and that missing SQLite would already cause a build failure, so I think you can keep this in your PR, it might be the wrong behaviour, but it's not really different from the existing behaviour, just making it clearer, so that's still an improvement.

Reply:
The existing makefiles had `-lsqlite3` hardcoded in them, so yeah, I think it was already required.

Reply:
Yeah, I think I'm wrong on that, it was already required. Let's keep it like that for now...